src/app/view/Limits/hooks/useOAuthClientsLimitExceeded.ts

```diff
@@ -0,0 +1,73 @@
+import { NavigationProp, ParamListBase } from '@react-navigation/native'
+import { useEffect, useState } from 'react'
+import type {
+  WebViewOpenWindowEvent,
+  WebViewNavigation
+} from 'react-native-webview/lib/WebViewTypes'
+
+import { useClient } from 'cozy-client'
+import Minilog from 'cozy-minilog'
+
+import {
+  OAUTH_CLIENTS_LIMIT_EXCEEDED,
+  interceptNavigation,
+  interceptOpenWindow,
+  oauthClientLimitEventHandler
+} from '/app/domain/limits/OauthClientsLimitService'
+
+const log = Minilog('⛔ OAuth Clients Limit Exceeded')
+
+const OAUTH_CLIENTS_LIMIT_EXCEEDED_URL_PATH = '/settings/clients/limit-exceeded'
+
+interface OAuthClientsLimitExceededState {
+  popupUrl: string | null
+  interceptNavigation: (request: WebViewNavigation) => boolean
+  interceptOpenWindow: (syntheticEvent: WebViewOpenWindowEvent) => void
+}
+
+export const useOAuthClientsLimitExceeded = (
+  navigation: NavigationProp<ParamListBase>
+): OAuthClientsLimitExceededState => {
+  const client = useClient()
+
+  const [popupUrl, setPopupUrl] = useState<string | null>(null)
+
+  const closePopup = (): void => {
+    setPopupUrl(null)
+  }
+
+  useEffect(() => {
+    const eventCallback = (href: string): void => {
+      if (client === null) {
+        log.error('Client is null, should not happen')
```

Comment:
if it's really impossible to happen maybe we should reflect it in the typing. if it's possible but very unlikely, shouldn't we logout() the user?